server/plugin.go

```diff
@@ -146,6 +156,32 @@ func (p *Plugin) OnPluginClusterEvent(c *plugin.Context, ev model.PluginClusterE
 	p.HandleClusterEvent(ev)
 }
 
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
```

Comment:
0/5 It would appear that in hi-traffic environments we should be using `MessageHasBeenPosted`, which is executing asynchronously and not `MessageWillBePosted` which is executed in-line.

I am very uneasy about tying up an RPC "connection" (state) to make other RPC, let alone HTTP requests. cc @agnivade for another opinion, it may be that the overhead here is not as significant as I think.

Reply:
`MessageWillBePosted` is necessary if a plugin needs to modify the incoming post, which in this case is needed.